fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -528,17 +694,63 @@ private FHIRRestOperationResponse doPatchOrUpdate(String type, String id, FHIRPa
                 }
             }
 
-            // write the resource back in case the interceptors modified it in some way
+            // capture the resource in case the interceptors modified it in some way
             newResource = event.getFhirResource();
 
+            // update the meta in the new resource. Use the version from the previous resource - this gets checked
+            // again under a database lock during the persistence phase and the request will be rejected if there's
+            // a mismatch (can happen when there are concurrent updates).
+            final com.ibm.fhir.model.type.Instant lastUpdated = com.ibm.fhir.model.type.Instant.now(ZoneOffset.UTC);
+            final int newVersionNumber = updateCreate ? 1 : Integer.parseInt(ior.getPrevResource().getMeta().getVersionId().getValue()) + 1;
+            newResource = FHIRPersistenceUtil.copyAndSetResourceMetaFields(newResource, newResource.getId(), newVersionNumber, lastUpdated);
+            
+            ior.setResource(newResource);
+            ior.setDeleted(isDeleted);
+            
+            // That's it for now - persistence is done later
+            return ior;
+        } finally {
+            // Restore the original request context.
+            FHIRRequestContext.set(requestContext);
+
+            log.exiting(this.getClass().getName(), "doUpdateMeta");
+        }
+    }
+    
+    @Override
+    public FHIRRestOperationResponse doPatchOrUpdate(String type, String id, FHIRPatch patch,
```

Comment:
`patch`, `ifMatchValue`, `searchQueryString`, `skippableUpdate` are not used in this method